runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java

```diff
@@ -495,8 +496,58 @@ public void unreachable(ProcessContext c) {
                   }
                 }));
 
-    thrown.expectCause(isA(CoderException.class));
-    thrown.expectMessage("cannot encode a null Long");
+    thrown.expectCause(
+        new ThrowableMessageMatcher<CoderException>(equalTo("cannot encode a null Long")));
+    thrown.expectMessage("Exception occurred while executing transform Generate step");
+    p.run();
+  }
+
+  @Test
+  public void testUserException() throws Exception {
+    Pipeline p = getPipeline();
+    p.apply("Step 1", Create.of(1))
+        .apply("Step 2", ParDo.of(new DoFn<Integer, Long>() {
+          @ProcessElement
+          public void processElement(ProcessContext c) {
+            throw new IllegalStateException("Custom exception element " + c.element());
+          }
+        }));
+
+    thrown.expectCause(
+        new ThrowableMessageMatcher<IllegalArgumentException>(equalTo("Custom exception "
```

Comment:
This generic doesn't actually exist, and even if it did, it wouldn't do anything, right?

Reply:
Removed generic